### DISCUSSION TITLE
git log --all --graph

### DISCUSSION THREAD

**Discussion by osvik** [jesseduffield/lazygit#2182] [Ideas]
It would be interesting to be able to view all branches in a graph and the graph inside lazygit. 

`git log --all --graph`

I have this as a custom command, but other programs allow us to see all commits and the branches tips and it's very useful.

**Comment by mark2185** [comment]
Have you tried going to the `Commits` panel (`4`)` and then pressing `+` twice in order to bring it up to `max` size?

**Reply by osvik** [reply]
Yes, but I can only see one branch at the time. Thank you.

**Reply by mark2185** [reply]
There's a `branchLogCmd` command in `config.yml` ([docs](https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md)) that might help you:
```
  branchLogCmd: 'git log --graph --color=always --abbrev-commit --decorate --date=relative --pretty=medium {{branchName}} --'
```

**Reply by osvik** [reply]
The file was empty, I've added the line above and didn't notice any difference.

**Reply by mark2185** [reply]
Did you add _my_ line (which is the default one) or did you add _your_ line which contains `--all`?

**Reply by osvik** [reply]
Your line.

**Reply by mark2185** [reply]
That explains why there was no difference, could you try it with your line from the original post?

**Reply by osvik** [reply]
No, still only shows one branch, not the same output I get with `git log --all --graph`.

I would suggest add a feature where we could see the different branches and commits from all branches. It's just a suggestion to improve lazygit.

**Reply by mark2185** [reply]
I apologize for the misinformation, I didn't know about these options.

The `branchLogCmd` affects the right panel's output when you select a branch from the `Local Branches` panel (`3`).

The `allBranchesLogCmd` affects the right panel's output when you go to the `Status` panel (`1`) and press `a` which stands for `show all branch logs`. Its default value contains `--graph --all`, among other flags.

**Reply by osvik** [reply]
Thank you! The default shows what I want. I didn't check the menu on this panel, indeed the "a" shows a log of all branches.

**Reply by mark2185** [reply]
Great! Feel free to mark the discussion as 'Solved' :)

**Reply by osvik** [reply]
I'm afraid I don't have that option in Github for this discussion, but thank you, the issue is solved.

**Reply by erlangparasu** [reply]
Thanks!
`4` `+` `+` works
